### PR TITLE
Note List Cell: New Metrics!

### PR DESCRIPTION
**Pull request by jleandroperez**
### Details:
In this PR we're several metrics affecting the Note Cell:

- The Accessory Icons's height now is dynamically adjusted to match the Title's Height. Capped between (16, 24) points
- We're now splitting Title and Body in two TextViews. This gives us a granular control over the number of lines in the Title / Body
- Several in-cell metrics have been fine tuned to match the mockups
- YES StackViews **all** over. Absolutely beautiful!!!

@aerych promise... last one regarding the Notes Cell!!!
**THANKYOU!!!**

Ref. #408

### Testing:
**Please** perform the following steps in these devices:
- [x] iPhone Xs
- [x] iPad Fullscreen (Portrait)
- [x] iPad Fullscreen (Landscape)
- [x] iPad Multitasking 1/3 (Landscape)
- [x] iPad Multitasking 1/2 (Landscape)
- [x] iPad Multitasking 2/3 (Landscape)

1. Log into Simplenote
2. Pin a note
3. Share a note
4. Verify the Note Row displays the icons on the right
5. Adjust the Font Size to the minimum, and verify the note row looks good
6. Repeat (5) but set it to the maximum
7. Please toggle Dark Mode and verify nothing breaks!

cc @SylvesterWilmott sir, mind taking a peek at the screenshots?

---

Regular
Min
Max
iPad Portrait
iPad Landscape
Dark

### CODE REVIEW

**Comment by SylvesterWilmott** [comment]
Looks good!

**Comment by jleandroperez** [comment]
Thank you Sly!

File: Simplenote/Classes/SPNoteTableViewCell.swift [hunk 1]
```diff
@@ -9,15 +9,15 @@ class SPNoteTableViewCell: UITableViewCell {
 
     /// Master View
     ///
-    @IBOutlet private var containerView: UIView!
+    @IBOutlet private var titleTextView: UITextView!
 
     /// Accessory StackView
     ///
-    @IBOutlet private var accessoryStackView: UIStackView!
+    @IBOutlet private var bodyTextView: UITextView!
```
Comment (aerych) [inline Simplenote/Classes/SPNoteTableViewCell.swift]: I'm curious about the choice of UITextView vs UILabel. Seems like these are just for display.  UILabel is lighter weight, supports attributed strings, doesn't have its own scroll view hierarchy embedded, and is mostly (completely?) configurable from the xib.

Comment (aerych) [reply Simplenote/Classes/SPNoteTableViewCell.swift]: Simulator Screen Shot - iPad Pro (9 7-inch) - 2019-09-25 at 16 49 25
I think UILabel's might be more #00ff00 as well.

**Comment by SylvesterWilmott** [comment]
Thanks @jleandroperez !

Looks great, Can we truncate the text labels using a `...`?

File: Simplenote/Classes/SPNoteListViewController.m [hunk 2]
```diff
@@ -221,7 +221,9 @@ - (void)updateRowHeight {
     CGFloat topTextViewPadding = verticalPadding;
 
     CGFloat numberLines = [[Options shared] numberOfPreviewLines];
-    CGFloat lineHeight = [@"Tommy" sizeWithAttributes:@{NSFontAttributeName: [UIFont preferredFontForTextStyle:UIFontTextStyleHeadline]}].height;
+    CGFloat lineHeight = [@"Tommy" sizeWithAttributes:@{
```
[the remark this answers is not in the document]
Comment (jleandroperez) [reply Simplenote/Classes/SPNoteListViewController.m]: Honestly... I don't know!! LOL!!! 

Thank you!!!

**Comment by jleandroperez** [comment]
@aerych (Nearly) all comments addressed. I took a decent stab at replacing UITextView <> UILabel, but in the end I couldn't match the way things look like, at a pixel perfect level.

Let's keep it _for now_, I'll add a note to revisit this later on!!

@SylvesterWilmott Truncation was *alledgedly* enabled, but indeed, wasn't working. Patched up now, sample screenshot below. Thank you!!!

---

Simulator Screen Shot - iPhone X - 2019-09-26 at 13 38 31

**Comment by jleandroperez** [comment]
I'm tracking a list of (related) tech debt tasks here: #408. Just added a note in that section!!

Thank you @aerych !!!!

**Pull request merged**